### DISCUSSION TITLE
How to use baseURL in rc.11??

### DISCUSSION THREAD

**Discussion by jprataciertocom** [nuxt/nuxt#16713] [Questions] [answered]
Since Nuxt rc.7 the NUXT_PUBLIC environment variables and base URLs are not set correctly in runtime as works in the rc.6.

Description:

**package.json**
```
"devDependencies": {
    "@babel/core": "^7.18.13",
    "@babel/preset-env": "^7.18.10",
    "@commitlint/cli": "^17.1.2",
    "@commitlint/config-conventional": "^17.1.0",
    "@nuxt/test-utils": "^0.2.2",
    "@nuxt/types": "^2.15.8",
    "@nuxtjs/eslint-config-typescript": "^11.0.0",
    "@nuxtjs/eslint-module": "^3.1.0",
    "@nuxtjs/svg": "^0.4.0",
    "@pinia/nuxt": "^0.4.1",
    "@pinia/testing": "^0.0.14",
    "@rollup/plugin-alias": "^3.1.9",
    "@types/jest": "^29.0.0",
    "@vitejs/plugin-vue": "^3.0.3",
    "@vitest/coverage-c8": "^0.23.0",
    "@vue/test-utils": "^2.0.2",
    "@vueuse/nuxt": "9.1.1",
    "axios": "^0.27.2",
    "cross-env": "^7.0.3",
    "eslint": "^8.23.0",
    "eslint-config-airbnb-base": "15.0.0",
    "eslint-import-resolver-typescript": "^3.5.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-vue": "^9.4.0",
    "jsdom": "^20.0.0",
    "lint-staged": "^13.0.3",
    "nuxt": "3.0.0-rc.11",
    "pinia": "^2.0.21",
    "postcss": "8.4.16",
    "prettier": "^2.7.1",
    "sass": "1.54.8",
    "sass-loader": "^13.0.2",
    "typescript": "^4.8.2",
    "vitest": "^0.23.0"
  },
    "dependencies": {
    "@apollo/client": "^3.6.9",
    "@babel/plugin-transform-runtime": "^7.18.10",
    "@nuxt/typescript-build": "^2.1.0",
    "@nuxt/webpack-builder": "3.0.0-rc.11",
    "@storyblok/vue": "^6.2.1",
    "@typescript-eslint/eslint-plugin": "^5.36.1",
    "@typescript-eslint/parser": "^5.36.1",
    "@vue/apollo-composable": "^4.0.0-alpha.19",
    "eslint-config-airbnb-typescript": "^17.0.0",
    "graphql": "^16.6.0",
    "postcss-loader": "^7.0.1",
    "vue-apollo": "^3.1.0"
  }
```

**nuxt.config.ts**
```
apiSecret: '',
  public: {
    ENV: 'local',
    API_ENDPOINT: 'http://no-variable',
    VERSION: packageJson.version,
    STORYBLOK_API_TOKEN: '',
    GTM_API_TOKEN: '',
    OPTIMIZE_TOKEN: '',
    WEB_SOCKET_URL: '',
    CDN_URL: '',
  },
}
```
  
**With the rc.6, the base URL and the rest of the API Secrets were injected correctly.**
Now with the rc.11, the baseUrl and Api Secrets are not injected, only if we access the main path, and then it redirects to the base URL. 

Example:
- We inject a baseURL -> '/base/urls/' in runtime as we do with the rc.6
- If we go to "https://mainPath.com/base/urls" the base URL and API secrets are not injected

Now we add the configuration for baseURL as https://v3.nuxtjs.org/api/composables/use-runtime-config/#appbaseurl and the base config in the nuxt.config.ts file:
```
app: {
  baseURL: '/base/urls/',
},
```

- If we go to "https://mainPath.com/base/urls" the base URL is injected correctly in the HTML script routes but API secrets are not injected. -> **_NETWORK 200 for the "https://mainPath.com"_**
- If we go to "https://mainPath.com" the browser redirects to "https://mainPath.com/base/urls" and the API secrets are correctly injected and the website is shown correctly -> **_NETWORK return 404 for the "https://mainPath.com"_ and overwrite the main path to "https://mainPath.com/base/urls"**

What changed since the rc.6??

**Comment by danielroe** [comment, marked as answer]
I wonder if you have `ssr: false` and are statically rendering your index page as a result? See https://github.com/nuxt/framework/issues/7826 and pending fix https://github.com/nuxt/framework/pull/7831.

**Reply by jprataciertocom** [reply]
Yes, I use `ssr: false`. With the https://github.com/nuxt/framework/issues/7826 hotfix, it works.
I can await the pending fix.

Thank's for your help!!
